demos/browser/app/meetingV2/video/RemoteVideoManager.ts

```diff
@@ -60,24 +65,46 @@ export default class VideoPreferenceManager {
       this.attendeeIdToVideoPreference.get(attendeeId).degradationPreference = preference
     } else {
       this.attendeeIdToVideoPreference.set(attendeeId, new VideoPreference(
-        attendeeId, VideoPreferenceManager.DefaultVideoTilePriority, VideoPreferenceManager.DefaultVideoTileTargetDisplaySize, preference));
+        attendeeId, RemoteVideoManager.DefaultVideoTilePriority, RemoteVideoManager.DefaultVideoTileTargetDisplaySize, preference));
     }
     this.updateDownlinkPreference();
   }
 
+  supportsRemoteVideoPreferences(): boolean {
+    return (this.downlinkPolicy instanceof VideoPriorityBasedPolicy);
+  }
+
   private updateDownlinkPreference(): void {
     if (this.attendeeIdToVideoPreference.size === 0) {
       // Preserve default behavior if no preferences have been set yet
       this.logger.info('No video preferences set yet, not updating');
       return;
     }
 
-    const videoPreferences = VideoPreferences.prepare();
-    for (const [attendeeId, preference] of this.attendeeIdToVideoPreference.entries()) {
-      if (this._visibleAttendees.includes(attendeeId)) {
-          videoPreferences.add(preference);
-      }
+    if (this.downlinkPolicy instanceof VideoPriorityBasedPolicy) {
+        const videoPreferences = VideoPreferences.prepare();
+        for (const [attendeeId, preference] of this.attendeeIdToVideoPreference.entries()) {
+            if (this._visibleAttendees.includes(attendeeId)) {
+                videoPreferences.add(preference);
+            }
+        }
+        this.downlinkPolicy.chooseRemoteVideoSources(videoPreferences.build());
+    } else {
+        // Just convert this list to  the simpler 'VideoSource' class used by all highest policy
+        let videoSources: VideoSource[] = [];
+        for (const attendeeId of this.attendeeIdToVideoPreference.keys()) {
```

Comment:
Can you remind me where we build this for non-priority downlink policy?

Reply:
It's this function here


```
  ensureDefaultPreferences(videoSources: VideoSource[]): void {
    this.logger.info(`Available remote video sources changed: ${JSON.stringify(videoSources)}`);
    for (const source of videoSources) {
        if (!this.attendeeIdToVideoPreference.has(source.attendee.attendeeId)) {
          this.attendeeIdToVideoPreference.set(
            source.attendee.attendeeId, 
            new VideoPreference(source.attendee.attendeeId, RemoteVideoManager.DefaultVideoTilePriority, RemoteVideoManager.DefaultVideoTileTargetDisplaySize))
        }
    }
  }
  ```